### PR TITLE
[platform && device]update centec e582 platform device config files and drivers, update sai to 1.2.4

### PR DESCRIPTION
**Pull request by yangbashuang**
**- What I did**
1. update centec e582 platform device config files and drivers
2. update sai to 1.2.4
**- How I did it**
update centec e582 platform device config files and drivers, update sai to 1.2.4
**- How to verify it**
1. build success
2. startup and init success
**- Description for the changelog**

update centec e582 platform device config files and drivers:
1. fix e582 driver bugs
2. support led control
3. support psuutil
4. support sfputil
5. update e582 device config file
update sai:
1. update sai to 1.2.4

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -77,3 +77,6 @@
 [submodule "platform/broadcom/sonic-platform-modules-mitac"]
 	path = platform/broadcom/sonic-platform-modules-mitac
 	url = https://github.com/MiTAC-EBU/sonic-platform-modules-mitac.git
+[submodule "platform/centec/sonic-platform-modules-e582"]
+	path = platform/centec/sonic-platform-modules-e582
+	url = https://github.com/CentecNetworks/sonic-platform-modules-e582
```
Comment (lguohan) [inline .gitmodules]: hi as we have discussed in community meeting, new platform module should be in tree. can you pull the source code into sonic-buildimage. Thanks.

File: platform/centec/one-image.mk [hunk 2]
```diff
@@ -3,6 +3,16 @@
...
 $(SONIC_ONE_IMAGE)_DOCKERS += $(SONIC_INSTALL_DOCKER_IMAGES)
+
+ifeq ($(SONIC_CONFIG_DEBUG),y)
+GDB = gdb
+VIM = vim
+OPENSSH = openssh-client
+SSHPASS = sshpass
+STRACE = strace
+$(SONIC_ONE_IMAGE)_DBG_PACKAGES += $(GDB) $(VIM) $(OPENSSH) $(SSHPASS) $(STRACE)
+endif
```
Comment (lguohan) [inline platform/centec/one-image.mk]: this is not centec platform specific. Can you remove from this PR. You can create a separate PR this DEBUG feature.

File: platform/centec/rules.mk [hunk 3]
```diff
@@ -1,5 +1,5 @@
+include $(PLATFORM_PATH)/platform-modules-centec-e582.mk
 include $(PLATFORM_PATH)/sdk.mk
-include $(PLATFORM_PATH)/sai.mk
```
Comment (lguohan) [inline platform/centec/rules.mk]: if sai.mk file is no longer needed, can you remove that file?

File: .gitmodules (hunk 1 above)
Comment (yangbashuang) [reply .gitmodules]: thanks, i will close this PR and create a new PR after these changes are done.

File: platform/centec/one-image.mk (hunk 2 above)
Comment (yangbashuang) [reply platform/centec/one-image.mk]: thanks, i will close this PR and create a new PR after these changes are done.

File: platform/centec/rules.mk (hunk 3 above)
Comment (yangbashuang) [reply platform/centec/rules.mk]: thanks, i will close this PR and create a new PR after these changes are done.

**Comment by yangbashuang** [comment]
thanks, i will close this PR and create a new PR after these changes are done.

**Pull request closed without merging**